src/libmongoc/src/mongoc/mongoc-topology-background-monitoring.c

```diff
@@ -48,6 +48,12 @@ static BSON_THREAD_FUN (srv_polling_run, topology_void)
       }
 
       /* This will check if a scan is due. */
+      if (!mongoc_topology_should_rescan_srv (topology)) {
```

Comment:
Once we have determined that the topology does not require SRV polling, we can terminate the thread. The topology will remain Unknown until the first SDAM scan, which occurs independently of SRV polling.